File: backend/internal/dto/submission.go
package dto

import "github.com/google/uuid"

type SubmissionAnswerRequest struct {
	QuestionID uuid.UUID `json:"questionId"`
	Answer     string    `json:"answer"`
}

type SubmissionRequest struct {
    UserID uuid.UUID `json:"userId"`

    PracticeID uuid.UUID `json:"practiceId"`

    Answers []SubmissionAnswerRequest `json:"answers"`
}

File: frontend/src/types/submission.ts
export interface SubmissionAnswer {
    questionId: string;
    answer: string;
}

export interface SubmissionRequest {

    userId: string;

    practiceId: string;

    duration: number;

    answers: SubmissionAnswer[];
}

export interface SubmissionResult {

    score: number;

    correct: number;

    wrong: number;

    xpEarned: number;

    level: number;

    totalXP: number;

    duration: number;
}